src/core/tsi/s2a/record_protocol/s2a_crypter.cc

```diff
@@ -256,3 +266,227 @@ void check_half_connection(s2a_crypter* crypter, bool in_half_connection,
   GPR_ASSERT(half_connection->additional_data_size ==
              expected_additional_data_size);
 }
+
+/** This function returns the max number of bytes occupied by the nonce of a
+ *  TLS 1.3 record that is handled by |crypter|. The caller must not pass in
+ *  nullptr for |crypter|.
+ *  - crypter: an instance of s2a_crypter. **/
+static size_t s2a_max_aead_nonce_size(const s2a_crypter* crypter) {
+  GPR_ASSERT(crypter != nullptr);
+  /** If additional supported ciphersuites are added, then there may be
+   *  additional options for the return value. **/
+  return kEvpAeadMaxNonceLength;
+}
+
+/** This function increments the sequence field of |half_connection|. If the
+ *  sequence number overflows, then the function returns
+ *  GRPC_STATUS_OUT_OF_RANGE, and the channel must be closed. If
+ *  |half_connection| is nullptr, then the function returns
+ *  GRPC_STATUS_INTERNAL. Otherwise, the function returns GRPC_STATUS_OK.
+ *  - half_connection: an instance of s2a_half_connection. **/
+grpc_status_code increment_sequence(s2a_half_connection* half_connection) {
+  if (half_connection == nullptr) {
+    return GRPC_STATUS_INTERNAL;
+  }
+
+  half_connection->sequence += 1;
+  /** Check whether the sequence number has overflowed. **/
+  if (half_connection->sequence == 0) {
```

Comment:
I think it's fine to use the current check since uint64_t will always occupy 8 bytes across different platforms.

Reply:
Sounds good, left it as is.